src/vma/dev/time_converter.cpp

```diff
@@ -156,11 +156,14 @@ ts_conversion_mode_t time_converter::update_device_converters_status(net_device_
 
 void time_converter::clean_obj()
 {
-	set_cleaned();
+	if (is_cleaned()) {
+		return ;
+	}
 
-	if (m_timer_handle && g_p_event_handler_manager->is_running()) {
-		g_p_event_handler_manager->unregister_timers_event_and_delete(this);
+	set_cleaned();
+	if (g_p_event_handler_manager->is_running()) {
```

Comment:
What about cheking that g_p_event_handler_manager != NULL?

Reply:
Stopping internal thread and releasing `g_p_event_handler_manager` are two independant steps. Releasing of `g_p_event_handler_manager` is done at the end.
So I think checking `g_p_event_handler_manager` for NULL is needless. It is a global object that has strong creating/releasing order.

Reply:
What about resets_global() done in fork or daemon flows?